tests/regressions/error_in_return_type.c3

```diff
@@ -1,8 +1,6 @@
 typedef [T] Span : {}
 
-// Hmm, the error is a bit misleading...
-// The overload resolution fails due to SFINAE not parameter count.. but this is
-// also kinda the point of SFINAE, if we fixed this `EnableIf` would break.
+// TODO: the line number still isn't right here
 function [T] foo : ()
     -> Span = {}
```

Comment:
I swear I fixed that in a previous commit, before your big refactor. I just added a separate try/catch block around the code that parses the return type, so that it would wrap the error with the return type's location info.

Reply:
Yeah, this defo worked perfectly before the refactor.

I introduced this problem by allowing functions to sit around in an unresolved state. This allows us to resolve everything simultaneously but means that we only know about these errors after the program tries to resolve an exact function signature (by this point in time we've lost the exact line numbers).

I've made this much better (but not perfect) in e49243b608b8044df3c5ef6f5b52022b46815f21 by wrapping any errors with the line number of the function signature (note: in this example that means the line number is off-by-one).

Reply:
Right. We don't resolve inside `graphene_parser.py` anymore. I guess we'll have to start passing the tokens too.